core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -279,4 +289,45 @@ public void close() throws IOException {
           }
         });
   }
+
+  @FunctionalInterface
+  interface CreateTasksFunction<T extends ScanTask> {
```

Comment:
I could use `BiFunction` here but the definition would be bulky so I decided to add a functional interface just to shorten lines.

Reply:
I think it's more readable to use functional interface. +1 for this.